frontend/src/routes/flappy/script.js

```diff
@@ -0,0 +1,169 @@
+import { getCookie } from '../../components/storage/script.js';
+import { get_user } from '../../components/user/script.js';
+
+let userCookie = getCookie('user');
+let userElement = document.querySelector('#user_username');
+let isConnected = false;
+
+if (userCookie)
+{
+	const user = JSON.parse(userCookie);
+    userElement.innerText = `${user.username} is connected`;
+	isConnected = true;
+}
+else
+{
+	await get_user();
+	let userCookie = getCookie('user');
+
+	if (userCookie)
+	{
+		const user = JSON.parse(userCookie);
+		userElement.innerText = `${user.username} is connected`;
+		isConnected = true;
+	}
+	else
+	{
+		userElement.innerText = 'No user connected';
+		isConnected = false;
+	}
+
+}
+
+
+
+const canvas = document.getElementById('gameCanvas');
+const ctx = canvas.getContext('2d');
+
+const GRAVITY = 0.2;
+const JUMP_STRENGTH = -5.5;
+const OBSTACLE_WIDTH = 60;
+const HOLE_HEIGHT = 200;
+const OBSTACLE_SPACING = 300;
+const HOLE_MIN_HEIGHT = 100;
+const HOLE_MAX_HEIGHT = 400;
+
+let player = {
+    x: 100,
+    y: canvas.height / 2,
+    width: 30,
+    height: 30,
+    velocityY: 0,
+    jump() {
+        this.velocityY = JUMP_STRENGTH;
+    },
+    update() {
+        this.velocityY += GRAVITY;
+        this.y += this.velocityY;
+        
+        if (this.y + this.height > canvas.height) {
+            this.y = canvas.height - this.height;
+            this.velocityY = 0;
+        } else if (this.y < 0) {
+            this.y = 0;
+            this.velocityY = 0;
+        }
+    },
+    draw() {
+        ctx.fillStyle = 'black';
+        ctx.fillRect(this.x, this.y, this.width, this.height);
+    }
+};
+
+let obstacles = [];
+let frameCount = 0;
+let gameRunning = false;
+let score = 0;
+let gameSpeed = 0;
+let nextObstacleX = canvas.width;
+
+function startGame() {
+    gameRunning = true;
+    obstacles = [];
+    score = 0;
+    player.y = canvas.height / 2;
+    player.velocityY = 0;
+	nextObstacleX = canvas.width;
+    requestAnimationFrame(gameLoop);
```

Comment:
best to use setInterval because, requestAnimationFrame depends on the Hz of the screen